src/lib/middleware/origin-middleware.ts

```diff
@@ -0,0 +1,27 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { IUnleashConfig } from '../types';
+
+export const originMiddleware = ({
+    getLogger,
+    flagResolver,
+}: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>) => {
+    const logger = getLogger('/middleware/origin-middleware.ts');
+    logger.debug('Enabling origin middleware');
+
+    return (req: Request, _: Response, next: NextFunction) => {
+        if (!flagResolver.isEnabled('originMiddleware')) {
+            return next();
+        }
+
+        const isUI = !req.headers.authorization;
+
+        if (isUI) {
+            logger.debug('UI request');
+        } else {
+            logger.debug('API request');
+            logger.debug({ userAgent: req.headers['user-agent'] });
```

Comment:
Let's collapse these into one log line

Reply:
Like this? https://github.com/Unleash/unleash/pull/7695/commits/fa6c7155275b97308be7051ce61d002e5767440d